packages/components/src/internal/components/assay/AssayWizardModel.ts

```diff
@@ -96,6 +99,24 @@ export class AssayWizardModel
         return this.attachedFiles.valueSeq().toList();
     }
 
+    getTotalAttachedFilesSize(): number {
```

Comment:
These methods can avoid proliferating `immutable` objects by just returning arrays. All the use cases either convert them to arrays or check for size neither of which require immutable structures.

```tsx
getAttachedFiles(): File[] {
    return this.attachedFiles.valueSeq().toArray();
}

getTotalAttachedFilesSize(): number {
    return this.attachedFiles.reduce((totalSize, file) => totalSize + file.size, 0);
}

getResultsFiles(): File[] {
    return this.resultsFiles.valueSeq().toArray();
}

getTotalResultsFilesSize(): number {
    return this.resultsFiles.reduce((totalSize, file) => totalSize + file.size, 0);
}
```